### ISSUE TITLE
11 July Homepage content programming (Recommended Extensions launch)

### ISSUE THREAD

**Issue by devaneymoz** [mozilla/addons-frontend#8199]
To set the homepage for the debut of the Recommended Extensions program, I've [documented](https://docs.google.com/document/d/1Mtlh_fzKd7G25XKk9aR77Mt6eJrFk5CMc7cb8V5tQGc/edit?usp=sharing) all the granular content/copy changes that we'll need to implement for the **11 July** push. 

@bobsilverberg @muffinresearch

**Comment by bobsilverberg** [comment]
@devaneymoz This has landed on dev, so you should be able to test it there. Note that not all of the guides pages add-ons have been marked as recommended on dev, which is why you will not see a recommended badge next to most of them. If you make one recommended (which involves changing the add-on's status and then submitting a new version), you should then see the badge appear.

**Comment by ioanarusiczki** [comment]
@bobsilverberg I verified this on -dev with FF67(Win10).  The page looks as mentioned by the [document](https://docs.google.com/document/d/1Mtlh_fzKd7G25XKk9aR77Mt6eJrFk5CMc7cb8V5tQGc/edit)
I added the badge for the last guideline to check on it 
https://addons-dev.allizom.org/en-US/firefox/guides/enhance-your-media-experience/

I should keep an eye on -stage too in case this goes to prod this week. 

I've noticed that the tooltip of the recommended badge has the last word `performance` on a separate line. I could file an issue to make the entire text be available on one line. 

I think it would look better but there's one case where I can't tell if it would fit all in there :

![desktop full screen](https://user-images.githubusercontent.com/33448286/60885026-2b128800-a257-11e9-9898-46643e08778a.png)

This is how it looks like when I resize the browser

![resized browser](https://user-images.githubusercontent.com/33448286/60884979-03bbbb00-a257-11e9-9d35-b6239bdb5369.png)

**Comment by bobsilverberg** [comment]
> @bobsilverberg I verified this on -dev with FF67(Win10). The page looks as mentioned by the [document](https://docs.google.com/document/d/1Mtlh_fzKd7G25XKk9aR77Mt6eJrFk5CMc7cb8V5tQGc/edit)
> I added the badge for the last guideline to check on it
> [addons-dev.allizom.org/en-US/firefox/guides/enhance-your-media-experience](https://addons-dev.allizom.org/en-US/firefox/guides/enhance-your-media-experience/)
> 
> I should keep an eye on -stage too in case this goes to prod this week.
> 

Yes, the changes will land on stage today with the tag.

> I've noticed that the tooltip of the recommended badge has the last word `performance` on a separate line. I could file an issue to make the entire text be available on one line.
>

This is a product of the browser. There is nothing in our code that controls where this message breaks. For example, on Chrome, the message breaks after the word "meet". I don't believe there is anything we can do about this, other than make the entire message shorter, but there's still no guarantee of how Firefox will choose to render the message.

**Comment by bobsilverberg** [comment]
>     * The `Popular Themes` are 4 on a row. The `Recommended Themes` are 3 on the row.  I don't know if that's intended but my guess is that there should be only 3.
>

You are correct, there should only be 3 themes there. Can you please file an issue for that?
 
>     * The last section is not mentioned in the document `Change the way Firefox looks with themes.` but I believe that is to be kept as they redirect correctly to search results.

Yes, that section should remain as is.

**Comment by ioanarusiczki** [comment]
@bobsilverberg Looks good on stage - FF68 (Win10) and Android 8.0 

One note for stage: the `Default Bookmark Folder` extension is not available in 
https://addons.allizom.org/en-US/firefox/guides/organize-tabs-and-bookmarks/

I can't tell if this would affect or not AMO prod. 

![addon missing](https://user-images.githubusercontent.com/33448286/60969633-b1e36580-a328-11e9-9196-cc9fb689870c.png)

**Comment by ioanarusiczki** [comment]
@bobsilverberg 
One more question about the shelves :) 
In the document `Recommended extensions` refers to  
https://addons.mozilla.org/en-US/firefox/search/?featured=true&type=extension
the current link on stage is 
https://addons.allizom.org/en-US/firefox/search/?recommended=true&sort=random&type=extension

In the document `Popular extensions` refers to  
https://addons.mozilla.org/en-US/firefox/search/?featured=true&sort=users&type=extension
and the current link on stage is 
https://addons.allizom.org/en-US/firefox/search/?recommended=true&sort=users&type=extension

Basically, both shelves will display the recommended add-ons but the first has the `sort` random and the popular theme collection has the `sort` by users which makes information a bit redundant. 
On the other hand, if the number of recommended add-ons is consistent there should be no problems.

**Comment by bobsilverberg** [comment]
> One note for stage: the `Default Bookmark Folder` extension is not available in
> [addons.allizom.org/en-US/firefox/guides/organize-tabs-and-bookmarks](https://addons.allizom.org/en-US/firefox/guides/organize-tabs-and-bookmarks/)
> 
> I can't tell if this would affect or not AMO prod.

As long as the add-on exists on prod, which I assume it does as that's probably where the guid comes from, then there should be no issues with this.

**Comment by bobsilverberg** [comment]
> @bobsilverberg
> One more question about the shelves :)
> In the document `Recommended extensions` refers to
> [addons.mozilla.org/en-US/firefox/search/?featured=true&type=extension](https://addons.mozilla.org/en-US/firefox/search/?featured=true&type=extension)
> the current link on stage is
> [addons.allizom.org/en-US/firefox/search/?recommended=true&sort=random&type=extension](https://addons.allizom.org/en-US/firefox/search/?recommended=true&sort=random&type=extension)

This is the correct link. The document uses old links which are incorrect.

> 
> In the document `Popular extensions` refers to
> [addons.mozilla.org/en-US/firefox/search/?featured=true&sort=users&type=extension](https://addons.mozilla.org/en-US/firefox/search/?featured=true&sort=users&type=extension)
> and the current link on stage is
> [addons.allizom.org/en-US/firefox/search/?recommended=true&sort=users&type=extension](https://addons.allizom.org/en-US/firefox/search/?recommended=true&sort=users&type=extension)
>

This is also the correct link. Featured no longer exists, and for this type of link we do want the sort to be `users`.
 
> Basically, both shelves will display the recommended add-ons but the first has the `sort` random and the popular theme collection has the `sort` by users, which makes information a bit redundant.
> On the other hand, if the number of recommended add-ons is consistent there should be no problems.

Our intention is for each link to lead to a differently sorted list, so I think this is fine.

**Comment by ioanarusiczki** [comment]
@bobsilverberg Ok, this issue also needs a  sanity check after the push.

**Comment by bobsilverberg** [comment]
> @bobsilverberg Ok, this issue also needs a sanity check after the push.

Agreed!
